test/trento/settings_test.exs

```diff
@@ -128,402 +129,384 @@ defmodule Trento.SettingsTest do
+             %{
+               correlation_scenario: correlation_scenario,
+               correlation_id: correlation_id
+             } do
+          insert_list(5, :host)
+          insert(:host, deregistered_at: DateTime.to_iso8601(Faker.DateTime.backward(2)))
+
+          case correlation_scenario do
+            :with_correlation ->
+              expect(
+                Trento.Commanded.Mock,
+                :dispatch,
+                5,
+                fn %CompleteSoftwareUpdatesDiscovery{},
+                   [correlation_id: ^correlation_id, causation_id: ^correlation_id] ->
+                  :ok
+                end
+              )
 
-      for %{change_submissions: change_submissions, errors: errors} <- change_settings_scenarios do
-        change_submissions
-        |> List.wrap()
-        |> Enum.each(fn change_submission ->
-          assert {:error, %{errors: ^errors}} =
-                   Settings.change_suse_manager_settings(change_submission)
-        end)
-      end
-    end
+            :without_correlation ->
+              expect(
+                Trento.Commanded.Mock,
+                :dispatch,
+                5,
+                fn %CompleteSoftwareUpdatesDiscovery{} -> :ok end
+              )
+          end
 
-    test "should support partial change of suse manager settings" do
-      %{
-        url: initial_url,
-        username: _initial_username,
-        password: _initial_password,
-        ca_cert: initial_ca_cert,
-        ca_uploaded_at: initial_ca_uploaded_at
-      } =
-        insert_software_updates_settings(
-          ca_cert: Faker.Lorem.sentence(),
-          ca_uploaded_at: DateTime.utc_now()
-        )
+          settings = %{
+            url: "https://valid.com",
+            username: Faker.Internet.user_name(),
+            password: Faker.Lorem.word(),
+            ca_cert: build(:self_signed_certificate)
+          }
 
-      change_submission = %{
-        username: new_username = "new_username",
-        password: new_password = "new_password"
-      }
+          case inspect(@operation) do
+            "&Trento.Settings.save_suse_manager_settings/1" ->
+              :ok
 
-      assert {:ok,
-              %{
-                url: ^initial_url,
-                username: ^new_username,
-                password: ^new_password,
-                ca_cert: ^initial_ca_cert,
-                ca_uploaded_at: ^initial_ca_uploaded_at
-              }} = Settings.change_suse_manager_settings(change_submission)
-    end
+            "&Trento.Settings.change_suse_manager_settings/1" ->
+              expect(Trento.SoftwareUpdates.Discovery.Mock, :clear, 1, fn -> :ok end)
 
-    test "should properly update ca_cert and its upload date when a new cert is provided" do
-      now = DateTime.utc_now()
+              # there is a path dependence on needing to save first
+              # before trying to change settings
```

Comment:
this comment can be safely removed since now we're relying on `insert_software_updates_settings`